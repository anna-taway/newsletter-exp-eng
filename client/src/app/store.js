import { configureStore } from "@reduxjs/toolkit";
import registrationReducer from "../features/userRegistration/userRegistrationSlice";
import productsReducer from '../features/products/productsSlice';


const store = configureStore({
	reducer: {
		registration: registrationReducer,
		products: productsReducer
	},
});

export default store;