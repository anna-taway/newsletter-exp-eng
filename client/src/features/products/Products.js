import React, { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { getProducts } from './productsAction'

import { makeStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Grid from '@material-ui/core/Grid';

const useStyles = makeStyles((theme) => ({
  card: {
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
  },
  cardMedia: {
    paddingTop: '56.25%', // 16:9
  },
  cardContent: {
    flexGrow: 1,
  }
}));

const Products = () => {
  const classes = useStyles();
  const dispatch = useDispatch()
  const products = useSelector((state) => state.products)

  useEffect(() => {
    dispatch(getProducts())
  }, [dispatch])


  return (
    <React.Fragment>
        <Grid container spacing={4}>
          {products.listProducts.map( (product) => (
            <Grid item key={product.id} xs={12} sm={6} md={4}>
              <Card className={classes.card}>
                <CardMedia
                  className={classes.cardMedia}
                  image="https://source.unsplash.com/random"
                  title="Image title"
                />
                <CardContent className={classes.cardContent}>
                  <Typography gutterBottom variant="h5" component="h2">
                    Id: {product.id}
                  </Typography>
                  <Typography>
                    Name: {product.name}                    
                  </Typography>
                  <Typography>
                    Color: {product.color}                   
                  </Typography>
                  <Typography>
                    Pantone_value: {product.pantone_value}                    
                  </Typography>
                </CardContent>
                <CardActions>
                  <Button size="small" color="primary">
                    View
                  </Button>
                  <Button size="small" color="primary">
                    Edit
                  </Button>
                </CardActions>
              </Card>
            </Grid>
          ))}
        </Grid>
      </React.Fragment>
  )
}

export default Products