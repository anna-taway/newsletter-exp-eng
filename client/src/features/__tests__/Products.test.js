import React from "react";
import Products from "../products/Products";
import { create } from "react-test-renderer";
import { createMemoryHistory } from 'history'
import { Router } from 'react-router-dom'

import { Provider } from 'react-redux';
import store from '../../app/store';

describe("Products component", () => {

  test("Products component", () => {
    const history = createMemoryHistory()
    const products = create(
        <Provider store={store}>
            <Router history={history}>
                <Products />
            </Router>
        </Provider>
    )
    expect(products.toJSON()).toMatchSnapshot();
  });

});