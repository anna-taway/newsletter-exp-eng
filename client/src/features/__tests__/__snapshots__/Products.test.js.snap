// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products component Products component 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4"
/>
`;
